lib/constants.js

```diff
@@ -1,4 +1,11 @@
 module.exports = {
+  state: {
+    BEFORE_HEAD: 1,
+    IN_HEAD: 2,
+    IN_BODY: 3,
+    BEFORE_CHUNK: 4,
+    IN_CHUNK: 5
+  },
```

Comment:
This is basically a condensed version of the previous `readingEnum` and the various `HTTPClientRequest` flags.

Reply:
good insights there, thanks

Reply:
Support for `http.request(url[, options][, callback])` added. Also, new unit tests to cover that.